### PR TITLE
Additional test case along with windows issue fix for .import use case handling for Python

### PR DESCRIPTION
**Pull request by pandurangpatil**
Additional test case along with windows issue fix for .import use case handling for Python.

### CODE REVIEW

File: joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/PythonTypeRecovery.scala [hunk 1]
```diff
@@ -62,8 +62,8 @@ class RecoverForPythonFile(
           val fileName = path.file.name.headOption.getOrElse("").stripPrefix(root)
           val sep      = Matcher.quoteReplacement(JFile.separator)
           // The below gives us the full path of the relative "."
-          if (fileName.contains(sep))
-            fileName.substring(0, fileName.lastIndexOf(sep)).replaceAll(sep, ".")
+          if (fileName.contains(java.io.File.separator))
```
Comment (DavidBakerEffendi) [inline joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/PythonTypeRecovery.scala:65]:
```suggestion
          if (fileName.contains(JFile.separator))
```

File: joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/PythonTypeRecovery.scala [hunk 2]
```diff
@@ -62,8 +62,8 @@ class RecoverForPythonFile(
...
+            fileName.substring(0, fileName.lastIndexOf(java.io.File.separator)).replaceAll(sep, ".")
```
Comment (DavidBakerEffendi) [inline joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/PythonTypeRecovery.scala:66]:
```suggestion
            fileName.substring(0, fileName.lastIndexOf(JFile.separator)).replaceAll(sep, ".")
```

File: joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/passes/frontend/XTypeRecovery.scala [hunk 3]
```diff
@@ -779,7 +779,9 @@ abstract class RecoverForXCompilationUnit[CompilationUnitType <: AstNode](
   ): Unit = {
     baseTypes
       .map(t => s"$t.${f.canonicalName}")
-      .flatMap(p => cpg.method.fullNameExact(p))
+      .flatMap(p => {
```
Comment (DavidBakerEffendi) [inline joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/passes/frontend/XTypeRecovery.scala:782]: Seems like this is mostly unchanged, could roll this back

File: joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/PythonTypeRecovery.scala (hunk 1 above)
Comment (pandurangpatil) [reply joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/PythonTypeRecovery.scala]: done

**Review by DavidBakerEffendi — APPROVED** [review]

**Review by fabsx00 — APPROVED** [review]

**Pull request merged**